Brazilian Portuguese Language Support 🇧🇷 (#553)
InnerSourceCommons/innersourcecommons.org: content/pt-br/learn/books/getting-started-with-innersource.md

---
type: "book"
title: 'Introdução ao InnerSource'
image: "/images/learn/books/getting-started-with-innersource-cover-thumb.jpg"
weight: 1
summary: InnerSource, o poderoso movimento para desenvolvimento de software de código aberto dentro das paredes de uma única organização, onde a "abertura" de um projeto se estende entre equipes dentro da empresa. O autor Andy Oram leva você ao mundo do InnerSource, começando pela revisão dos princípios que tornam o desenvolvimento de código aberto bem-sucedido e, em seguida, descrevendo como o InnerSource tem funcionado no PayPal.
book_author: Andy Oram
book_publish_date: Julho de 2015
book_publisher: O’Reilly Media
book_url: /documents/books/GettingStartedWithInnerSource.pdf
aliases:
    - /getting-started
---

> Bem-vindo ao InnerSource, o poderoso movimento para desenvolvimento de software de código aberto dentro das paredes de uma única organização, onde a "abertura" de um projeto se estende entre equipes dentro da empresa. Neste relatório, o editor e autor da O'Reilly, Andy Oram, leva você ao mundo do InnerSource, começando pela revisão dos princípios que tornam o desenvolvimento de código aberto bem-sucedido e, em seguida, descrevendo como o InnerSource tem funcionado no PayPal, o principal facilitador de comércio na Internet.

Reviewer: ```suggestion
Bem-vindo ao InnerSource, o poderoso movimento para desenvolvimento de software open source dentro das paredes de uma única organização, onde a "abertura" de um projeto se estende entre equipes dentro da empresa. Neste relatório, o editor e autor da O'Reilly, Andy Oram, leva você ao mundo do InnerSource, começando pela revisão dos princípios que tornam o desenvolvimento de open source bem-sucedido e, em seguida, descrevendo como o InnerSource tem funcionado no PayPal, o principal facilitador de comércio na Internet.
```

O caminho do PayPal para o InnerSource envolveu uma série de decisões corporativas em grande escala, que incluíram uma mudança consciente de ferramentas e cultura corporativa. Através do InnerSource, a empresa não apenas alcançou um desenvolvimento mais rápido e melhor qualidade, mas também criou um ambiente de cooperação entre equipes que incentivou os programadores a contribuir para qualquer projeto de desenvolvimento do PayPal.

Você aprenderá as vantagens específicas da estratégia InnerSource, incluindo:

* Desenvolvimento mais rápido: Os programadores utilizam testes unitários, cobertura de código e integração contínua para eliminar erros nas fases iniciais.
* Documentação completa: O código é documentado de forma mais adequada, tanto em comentários no próprio código quanto de forma mais informal em listas de discussão.
* Reutilização de código: Os programadores em toda a organização compreendem o código e a arquitetura dos módulos desenvolvidos por outras equipes.
* Colaboração entre equipes: As contribuições de membros fora da equipe são fluidas e raramente precisam ser reescritas.

> * Desenvolvimento com o GitHub: O GitHub mantém repositórios privados para projetos internos, bem como repositórios públicos para código de código aberto.

Reviewer: ```suggestion
* Desenvolvimento com o GitHub: O GitHub mantém repositórios privados para projetos internos, bem como repositórios públicos para código open source.
```

Este livro aberto está licenciado sob uma Licença Creative Commons (CC BY). Você pode baixar o ebook Introdução ao InnerSource gratuitamente em formato PDF (3,3 MB).

## Sobre o autor

> **Andy Oram** é editor na O’Reilly Media. Funcionário da empresa desde 1992, Andy atualmente se especializa em tecnologias de código aberto e engenharia de software. Seus trabalhos para a O’Reilly incluem os primeiros livros lançados por uma editora dos EUA sobre Linux, o título de 2001 Peer-to-Peer e o best-seller

Reviewer: ```suggestion
**Andy Oram** é editor na O’Reilly Media. Funcionário da empresa desde 1992, Andy atualmente se especializa em tecnologias open source e engenharia de software. Seus trabalhos para a O’Reilly incluem os primeiros livros lançados por uma editora dos EUA sobre Linux, o título de 2001 Peer-to-Peer e o best-seller
```
